nevermined-io/contracts
# Bringing v2.0.0 to `master`

- circuit compiles
- figuring it out ...
- trying to figure it out
- weird ...
- slowly figuring it out
- slowly figuring it out
- perhaps near solution
- proving seems to work
- others working
- access agreement now works
- lint
- Adding a test for checking distinct eventIds
- revert the revert
- assigning to Nevermined
- Adding first version of `NeverminedConfig`
- Adding Config + Documentation
- fulfill conditions
- added proxy method to nft transfer
- checking agreement
- first version working
- selling many nft tokens
- doc and linting
- refactor conditions
- lint
- cleanup
- Adapting tests
- Integrating marketplace fees validation in LockPaymentCondition
- fixing upgrade tests
- Adding NeverminedConfig upgradeability tests
- cleaning license message
- Removing ConfigStorageV1 for simplicity
- also check the token address in addition to price
- updating version
- fixing deploy
- updating version
- update yarn
- fixinf upgrade test
- Renaming INeverminedConfig to INVMConfig to avoid bytecode generation problem
- Adding v1.99.8 Changelog updates
- also erc 721 version is now working
- merge
- upgrade gnosis safe sdk
- seems to work
- oz upgrades totally broken
- problem with contracts having the same bytecode ...
- First draft of Nevermined contracts ABI documentation
- Adding documentation about Nevermined contracts ABI integration
- testing
- failing upgrade
- debug
- clean up
- lint
- deploying
- read external artifacts when deploying contracts
- fixing upgrade tests
- approving and executing upgrade transactions
- cleanup
- Allows to NFT holders (ERC1155 & 721) to burn their NFTs if they are holders Implements https://github.com/nevermined-io/internal/issues/200
- removing unused variable
- Added script for managing artifacts and upload to s3
- TransferConditions allow to transfer or lazy mint
- removing un-used vars
- adding owner check
- Removing burn limitation. Every holder can burn directly and not just through the `DIDRegistry`
- Create a registry json file for deployment in artifact repository
- Adding protection to `burn` methods
- Adding DID ownership validation before minting in transfer condition
- Add test action for uploading artifacts
- Add missing git dependencies
- upload_artifacts_s3.sh invocation from action
- typo
- Add zip dependency
- Adding NFT owner validation in ERC-721 Transfer minting condition
- Debug bash script
- Compile contracts when uploading artifacts
- Using yarn for npm
- Moving action to ubuntu
- cleanup
- Run commands as sudo
- Skip contracts test when uploading artifacts
- regenerate zkey
- upgrading ...
- compilation warnings
- change instance variable order
- remove only
- update rest of the packages
- removing old comments
- shipping just `development` artifacts
- deploying managers
- Adding v1.99.9 Changelog updates
- deploying conditions
- handling addesses when deploying
- first test working
- still nft tests left
- first nft test working
- Doc about how to upload artifacts to repo
- Added gh workflow upload example
- most nft tests working
- integration tests working
- working with upgraded version
- lint
- fixing tests
- change deployment of old contracts
- run tests after upgrading
- fixing script
- resetting marketplace fees
- cleanup reinit
- bumping to `v2.0.0-rc0`
- Adding v2.0.0-rc0 Changelog updates
- Adding artifacts repository reference
- try to get test working without generated key
- set toolchain
- remove submodule
- remove from npm
- trying to upload circuits
- working on docker image
- working on docker image
- working on docker image
- Adding Transfer NFT Conditions documentation
- using same parameter format
- Adding `getNFTDefaultAddress` function into Transfer Conditions to understand the default NFT Contract address used
- using v2.0.0-rc1
- Adding v2.0.0-rc1 Changelog updates
- Deployed v2.0.0-rc0 to alfajores and mumbai
- Alchemy is not supporting Kovan anymore, so we try to run external tests with Infura
- Removed contract table from README.md (replaced by artifact repository)
- Updated release process to include uploading the contracts to s3
- Ignore artifacts folder in git
- Ignoring artifacts folder
- coming back to alchemy
- Keep artifacts folder
- Keep artifacts folder
- using the right env variable name for kovan alchemy token
- removing dash from url
- forgot to add this
- updating this again
- updating this again
- fix function name
- updating
- adding check for circuit change
- fix lint
- found the location of variable use
- stopped using variables
- mostly working
- should be working
- Use moralis.io kovan archive endpoint
- Use moralis.io kovan archive endpoint
- typo
- bumping to v2.0.0-rc2
- Adding v2.0.0-rc2 Changelog updates
- Removing Maven stubs generation from workflows
- removing references to Java
- turn optimization back on
- update main dockerfile
- update main dockerfile
- update main dockerfile
- Adding v2.0.0-rc3 Changelog updates
- Adding v2.0.0-rc3 Changelog updates
- dedup fulfill for delegate
- adding coverage ...
- lint
- still some more coverage
- still some more coverage
- building circuit: failing properly ...
- added royalty scheme
- added royalty recipient interface
- first draft for sending to royalty recipient
- keeping circuits and artifacts folders in NPM package
- trying to get this running properly
- fixing path
- slither running
- fixing slither issues
- lint
- fixing this ...
- running mythril
- trying to fix
- changing governor account
- maybe this works
- create directory
- Adding v2.0.0-rc4 Changelog updates
- Fixing setup of contracts and NPM package generation
- linting issues
- restoring verifier.sol
- coming back
- Adding v2.0.0-rc5 Changelog updates
- standard royalties
- tests working
- adding more tests
- checking royalties works
- lint
- adding integration test
- lint
- trying to deploy
- thinking about it...
- first version of distributor
- distributing royalties
- lint
- adding curve royalties contract
- adding test
- fixing script
- refactoring deploy transactions
- upgrade polygon node to latest version
- Gracefully shut down node to prevent data corruption
- rename distributor
- distributing ether rewards
- adding docs
- remove this
- remove this
- cleanup
- forgot to add getter for royalty scheme
- test for getter
- bumping to `v2.0.0`

## Description

Add a description of your changes here.

## Is this PR related with an open issue?

Related to Issue #

## Types of changes

 
## Checklist:

- [ ] Follows the code style of this project.
- [ ] Tests Cover Changes

#### Funny gif

![Put a link of a funny gif inside the parenthesis-->]()
